src/App.tsx

```diff
@@ -1,27 +1,50 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [itemsPerPage, setItemPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const startEl = (currentPage - 1) * itemsPerPage + 1;
+  const endEl =
+    currentPage * itemsPerPage > items.length
+      ? items.length
+      : currentPage * itemsPerPage;
+
+  const onChangePage = (page: number) => {
+    if (page !== currentPage) {
+      setCurrentPage(page);
+    }
+  };
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {currentPage} (items {startEl} - {endEl} of {items.length})
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
             id="perPageSelector"
-            className="form-control">
+            className="form-control"
+            onChange={e => {
+              setItemPerPage(+e.target.value);
+              setCurrentPage(1);
+            }}
```

Comment:
Pls move the function inside `onChange` handler to a separate const and use as a callback here